### DISCUSSION TITLE
vpot acceleration

### DISCUSSION THREAD

**Discussion by markusschloesser** [markusschloesser/MackieC4_P3#101] [General]
https://github.com/markusschloesser/MackieC4_P3/blob/617996f6140447a24b9de86d0c583df7cb389537/wip/MackieC4/MackieC4.py#L253

I already did use acceleration! ;-)
The faster you turn, the higher the cc value gets, that's how you can scroll so quickly with `self.song().jump_by(cc_value)`.
One click is just 1 bar, but if rotated quickly the cc value will be a lot higher and therefore also the scolling.
Or did I misunderstood you? @BilldarBagdar

**Comment by BilldarBagdar** [comment]
Nope, you got it.  That's why I said I got carried away,  I was just mentioning we could also use scaling to alter the pace of the acceleration.  As I recall, the ramp from "tiny jumps" to "mega jumps" completes really fast, it's difficult to "maintain" a medium sized jump during acceleration, or to slow down the acceleration gradually as you approach your target.  There are basically two speeds "super slow" and "super fast", nothing in between.

The biggest step value would still be "15" but it would take twice as many steps to get there, for example.  (If your twist would go 180 degrees around the knob, now you can ramp to full jump size in like 18 degrees of "quick twist" (10% of the travel) or less, but with a scaling factor we could make that ramp take 20, 25 or even 50% of the travel before the jumps reach full size).

**Reply by markusschloesser** [reply]
> As I recall, the ramp from "tiny jumps" to "mega jumps" completes really fast, it's difficult to "maintain" a medium sized jump during acceleration, or to slow down the acceleration gradually as you approach your target. There are basically two speeds "super slow" and "super fast", nothing in between.

Really? My experience has been quite nice.
How do you perceive the scrolling in arrange mode with vpot 12? I experimented with a couple of scaling factors before settling with the current direct cc value

**Comment by markusschloesser** [comment]
oh also, I cherrypicked your commits from c4modelling branch into 2022_dev.
Because your branch doesn't have the latest things I did.

oh and https://github.com/markusschloesser/MackieC4_P3/issues/100
